proto/p4/config/p4info.proto

```diff
@@ -80,26 +78,29 @@ message Table {
   repeated int32 action_ids = 3;
   // 0 (default value) means that the table does not have a const default action
   int32 const_default_action_id = 4;
+  // a table may have a const default action, whose action parameter values can
+  // be changed at runtime. However, in most cases a table's action entry is set
+  // in the P4 program and cannot be changed, which is what this boolean flag
+  // indicates.
+  bool has_const_default_action_entry = 5;
   // P4 id of the "implementation" for this table (e.g. action profile id); 0
   // (default value) means that the table is a regular (direct) match table. As
   // of today, only action profiles are supported but other table
   // implementations may be added in the future
-  int32 implementation_id = 5;
+  int32 implementation_id = 6;
   // ids of the direct resources (if any) attached to this table; for now this
   // includes only direct counters and direct meters, but other resources may be
   // added in the future
-  repeated int32 direct_resource_ids = 6;
-  int64 size = 7;  // number of entries in table
+  repeated int32 direct_resource_ids = 7;
+  int64 size = 8;  // number of entries in table
+  bool with_entry_timeout = 9;  // entry ageing is enabled for table
 }
 
 message Action {
   Preamble preamble = 1;
   message Param {
-    // id is not really needed, as it can be inferred from the action id and the
-    // position of the parameter, but having it could be convenient.
-    int32 id = 1;
-    string name = 2;
-    int32 bitwidth = 3;
+    Preamble preamble = 1;
```

Comment:
I am not convinced that the notion of path really extends to action params and match fields (but all the other fields of `Preamble` do apply). We don't have to use `Preamble` here, I can revert to the old code.
Apart from the action param case, it seems that we are in disagreement over path vs. name. As I mentioned above, from our discussions I believe that you don't really have a need for the `name` (you basically just need the `path`). For you the `name` should just be the last component of the `path`. However I do have use cases in mind for this `name` alias. In particular I do believe the ability for the P4 programmer to choose an alias for a P4 object that will be independent of the P4 hierarchy and resilient to code-refactoring is important.